src/libultra/rmon/xldtob.c

```diff
@@ -1,7 +1,276 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/xldtob/_Ldtob.s")
+#define BUFF_LEN 0x20
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/xldtob/_Ldunscale.s")
+s16 _Ldunscale(s16*, _Pft*);
+void _Genld(_Pft*, u8, u8*, s16, s16);
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/xldtob/_Genld.s")
+const f64 digs[] = { 10e0L, 10e1L, 10e3L, 10e7L, 10e15L, 10e31L, 10e63L, 10e127L, 10e255L };
+
+/* float properties */
+#define _D0 0
+#define _DBIAS 0x3ff
+#define _DLONG 1
+#define _DOFF 4
+#define _FBIAS 0x7e
+#define _FOFF 7
+#define _FRND 1
+#define _LBIAS 0x3ffe
+#define _LOFF 15
+/* integer properties */
+#define _C2 1
+#define _CSIGN 1
+#define _ILONG 0
+#define _MBMAX 8
+#define NAN 2
+#define INF 1
+#define FINITE -1
+#define _DFRAC ((1 << _DOFF) - 1)
+#define _DMASK (0x7fff & ~_DFRAC)
+#define _DMAX ((1 << (15 - _DOFF)) - 1)
+#define _DNAN (0x8000 | _DMAX << _DOFF | 1 << (_DOFF - 1))
+#define _DSIGN 0x8000
+#if _D0 == 3
+#define _D1 2 /* little-endian order */
+#define _D2 1
+#define _D3 0
+#else
+#define _D1 1 /* big-endian order */
+#define _D2 2
+#define _D3 3
+#endif
+
+void _Ldtob(_Pft* args, u8 type) {
+    u8 buff[BUFF_LEN];
+    u8* ptr = buff;
+    u32 sp70;
+    f64 val = args->v.ld;
+    /* maybe struct? */
+    s16 err;
+    s16 nsig;
+    s16 exp;
+
+    s32 i;
```

Comment:
Remove the empty line between variable declarations
```suggestion
    s16 exp;
    s32 i;
```